File: Stack/0957-minimum-add-to-make-parentheses-valid/solution.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        stack <char> st;
        for(int i=0;i<s.length();i++){
            if(s[i]=='(') st.push('(');
            else if(s[i]==')'){
                if (!st.empty() and st.top()=='('){
                    st.pop();
                }
                else{
                    st.push(')');
                }
            }
        }
        int count =0;
        while(!st.empty()){
            count++;
            st.pop();
        }
        return count;
    }
};

File: Stack/0957-minimum-add-to-make-parentheses-valid/solution.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        stack = []
        for n in s:
            if n == '(':
                stack.append(n)
            elif n == ')':
                if stack and stack[-1]== '(':
                    stack.pop()
                else:
                    stack.append(')')
        return len(stack)
